mod_form.php

```diff
@@ -138,6 +151,35 @@ public function definition() {
             $mform->addElement('static', 'additionalactivitiestwo', '', get_string('additionalactivitiestwo', 'accredible'));
         }
 
+        // Load Accredible attribute keys.
+        $attributekeys = $attributekeysclient->get_attribute_keys();
+        if (isset($attributekeys)) {
+            $attributekeyschoices = array('' => 'Select a Custom Attribute') + $attributekeys;
+            // Hidden element to check if we should disable the "gradeattributekeyname" select.
+            $mform->addElement('hidden', 'attributekysnumber', 1);
+        } else {
+            $mform->addElement('hidden', 'attributekysnumber', 0);
+        }
+
+        $mform->addElement('checkbox', 'includegradeattribute', get_string('includegradeattributedescription', 'accredible'),
+            get_string('includegradeattributecheckbox', 'accredible'));
+        if (isset( $accrediblecertificate->includegradeattribute )) {
+            $mform->setDefault('includegradeattribute', 1);
+            $includegradewrapperhtml = '<div id="include-grade-select-container" class="hidden">';
+        } else {
+            $includegradewrapperhtml = '<div id="include-grade-select-container">';
+        }
+
+        $mform->addElement('html', $includegradewrapperhtml);
+        $mform->addElement('select', 'gradeattributegradeitemid', get_string('gradeattributegradeitemselect', 'accredible'),
+            $assigmentschoices, $inputstyle);
+        $mform->addElement('select', 'gradeattributekeyname', get_string('gradeattributekeynameselect', 'accredible'),
+            $attributekeyschoices, $inputstyle);
+        $mform->disabledIf('gradeattributekeyname', 'attributekysnumber', 'eq', 0);
+        $mform->addElement('static', 'emptygradeattributekeyname', '', get_string('emptygradeattributekeyname', 'accredible',
+            $dashboardurl));
+        $mform->addElement('html', '</div>');
+
```

Comment:
You also need to update `accredible_add_instance` and `accredible_update_instance` in `lib.php` to save the selected values for the DB record.

Reply:
Added